supervisor/shared/filesystem.c

```diff
@@ -92,16 +94,42 @@ void filesystem_flush(void) {
     supervisor_flash_flush();
 }
 
-void filesystem_writable_by_python(bool writable) {
+void filesystem_set_internal_writable_by_usb(bool writable) {
     fs_user_mount_t *vfs = &_internal_vfs;
 
-    if (!writable) {
+    filesystem_set_writable_by_usb(vfs, writable);
+}
+
+void filesystem_set_writable_by_usb(fs_user_mount_t *vfs, bool usb_writable) {
+    if (usb_writable) {
         vfs->flags |= FSUSER_USB_WRITABLE;
     } else {
         vfs->flags &= ~FSUSER_USB_WRITABLE;
     }
 }
 
+bool filesystem_is_writable_by_python(fs_user_mount_t *vfs) {
+    return (vfs->flags & FSUSER_CONCURRENT_WRITE_PROTECTED) == 0 ||
```

Comment:
if `FSUSER_CONCURRENT_WRITE_PROTECTED` is set to true, then this will always be true. I don't think that's what you meant to do here. Maybe `&&`?

Reply:
I don't think so. True means the bit is set. This tests that the bit is 0 and short circuits the subsequent check when the concurrent bit is 0 aka off. (It is a bit confusing because the Python API is "disable" and its inverted internally in shared-module to "enable".)

Reply:
`== 0` was staring me in the face but I missed it :)